### PR TITLE
[NOD-1191] Convert wire protocol to proto

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: netadapter/server/grpcserver/protowire/messages.proto [hunk 1]
```diff
@@ -3,11 +3,188 @@ package protowire;
...
+  bytes bytes = 1;
+}
+// AddressesMessage End
+
+// GetAddressesMessage start
+message GetAddressesMessage{
+  bool includeAllSubnetworks = 1;
+  SubnetworkID subnetworkID = 2;
+}
+// GetAddressesMessage end
```
Comment (svarogg) [inline netadapter/server/grpcserver/protowire/messages.proto:52]: Sometimes the start/end are Upper Case, sometimes they are lower case. Please be consistent

File: netadapter/server/grpcserver/protowire/message_request_relay_blocks.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package protowire
+
+import (
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+)
+
+func (x *KaspadMessage_RequestRelayBlocks) toWireMessage() (wire.Message, error) {
+	if len(x.RequestRelayBlocks.Hashes) > wire.MsgGetRelayBlocksHashes {
```
Comment (svarogg) [inline netadapter/server/grpcserver/protowire/message_request_relay_blocks.go:9]: Please rename `wire.MsgGetRelayBlocksHashes` to `MaxMsgRequestBlockHashes` or something (Also go over the other renamings you did and see if there are any similar errors. I'll check myself, but might miss something.

File: netadapter/server/grpcserver/protowire/message_request_transactions.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package protowire
+
+import (
+	"github.com/kaspanet/kaspad/wire"
+	"github.com/pkg/errors"
+)
+
+func (x *KaspadMessage_RequestTransactions) toWireMessage() (wire.Message, error) {
+	if len(x.RequestTransactions.Ids) > wire.MaxInvPerGetTransactionsMsg {
```
Comment (svarogg) [inline netadapter/server/grpcserver/protowire/message_request_transactions.go:9]: Get in `wire.MaxInvPerGetTransactionsMsg` should become Request

File: wire/msgrequestaddresses_test.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// license that can be found in the LICENSE file.
+
+package wire
+
+import (
+	"testing"
+)
+
+// TestGetAddresses tests the MsgRequestAddresses API.
+func TestGetAddresses(t *testing.T) {
```
Comment (stasatdaglabs) [inline wire/msgrequestaddresses_test.go:12]: Please rename this to TestRequestAddresses.

File: wire/msgrequestblocklocator_test.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package wire
+
+import (
+	"testing"
+
+	"github.com/kaspanet/kaspad/util/daghash"
+)
+
+// TestGetBlockLocator tests the MsgRequestBlockLocator API.
+func TestGetBlockLocator(t *testing.T) {
```
Comment (stasatdaglabs) [inline wire/msgrequestblocklocator_test.go:10]: Please rename this to TestRequestBlockLocator.

File: wire/msgrequestibdblocks_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+package wire
+
+import (
+	"testing"
+
+	"github.com/kaspanet/kaspad/util/daghash"
+)
+
+// TestGetBlocks tests the MsgRequestIBDBlocks API.
+func TestGetBlocks(t *testing.T) {
```
Comment (stasatdaglabs) [inline wire/msgrequestibdblocks_test.go:14]: Please rename this to TestRequestBlocks.

File: wire/msgrequestselectedtip_test.go [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+// license that can be found in the LICENSE file.
+
+package wire
+
+import (
+	"testing"
+)
+
+// TestGetSelectedTip tests the MsgRequestSelectedTip API.
+func TestGetSelectedTip(t *testing.T) {
```
Comment (stasatdaglabs) [inline wire/msgrequestselectedtip_test.go:12]: Please rename this to TestRequestSelectedTip.

**Pull request merged**